### PR TITLE
#19849 Fixes issue in theme selector

### PR DESCRIPTION
**Pull request by ivorpad**
Ref: https://github.com/dotCMS/core/issues/19849#issuecomment-770066510

### CODE REVIEW

File: src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts [hunk 1]
```diff
@@ -78,8 +83,12 @@ export class DotSiteSelectorComponent implements OnInit, OnChanges, OnDestroy {
                 });
         });
 
+        this.currentSite.subscribe((res) => {
+            console.log(res.hostname);
+        });
+
```
Comment (fmontes) [inline src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts:89]: remove.

File: src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts [hunk 2]
```diff
@@ -57,6 +58,10 @@ export class DotSiteSelectorComponent implements OnInit, OnChanges, OnDestroy {
         private dotEventsService: DotEventsService
     ) {}
 
+    get currentSite(): Observable<Site> {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts:61]:
```suggestion
    get currentSite$(): Observable<Site> {
```

I forgot about this but is a good practice to append a `$` to the properties that are `Observables`

File: src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts [hunk 3]
```diff
@@ -45,7 +45,8 @@ export class DotSiteSelectorComponent implements OnInit, OnChanges, OnDestroy {
 
     @ViewChild('searchableDropdown') searchableDropdown: SearchableDropdownComponent;
 
-    currentSite: Observable<Site>;
+    currentSiteSub: Subject<Site> = new Subject();
```
Comment (fmontes) [inline src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts:48]:
```suggestion
    currentSiteSub$: Subject<Site> = new Subject();
```

File: src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts [hunk 4]
```diff
@@ -105,9 +115,22 @@ export class DotThemeSelectorDropdownComponent
      * @memberof DotThemeSelectorDropdownComponent
      */
     siteChange(event: Site): void {
+        this.currentSiteIdentifier = event.identifier;
         this.setHostThemes(event.identifier);
     }
 
+    onShow(): void {
```
Comment (fmontes) [inline src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts:122]: Add docs.

File: src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts [hunk 5]
```diff
@@ -139,32 +162,25 @@ export class DotThemeSelectorDropdownComponent
...
+                });
+        }
     }
 
     private getFilteredThemes(filter = '', offset = 0): void {
         this.paginatorService.searchParam = filter;
+        this.setHostThemes(this.currentSiteIdentifier, this.currentOffset || offset);
+    }
 
+    private setHostThemes(identifier: string, offset: number = 0) {
```
Comment (fmontes) [inline src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts:180]:
```suggestion
    private setHostThemes(hostId: string, offset: number = 0) {
```

File: src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts [hunk 6]
```diff
@@ -70,6 +64,19 @@ export class DotThemeSelectorDropdownComponent
         }
     }
 
+    onHide(): void {
```
Comment (fmontes) [inline src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts:67]: docs.

File: src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts [hunk 7]
```diff
@@ -70,6 +64,19 @@ export class DotThemeSelectorDropdownComponent
...
+        if (this.value) {
+            this.siteService.getSiteById(this.value.hostId).subscribe((site) => {
```
Comment (fmontes) [inline src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts:69]: This needs a `take(1)`

**Review by fmontes — CHANGES_REQUESTED** [review]

File: src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts [hunk 8]
```diff
@@ -169,11 +173,7 @@ export class DotSiteSelectorComponent implements OnInit, OnChanges, OnDestroy {
         this.paginationService.filter = filter;
         this.paginationService.getWithOffset(offset).subscribe((items) => {
             this.sitesCurrentPage = [...items];
-            this.totalRecords = this.totalRecords || this.paginationService.totalRecords;
```
Comment (hmoreras) [inline src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts:172]: this is not needed ?

File: src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts [hunk 9]
```diff
@@ -186,6 +186,15 @@ export class DotSiteSelectorComponent implements OnInit, OnChanges, OnDestroy {
         this.change.emit(site);
     }
 
+    /**
+     * Set default site as current
+     *
+     * @memberof DotSiteSelectorComponent
+     */
+    setCurrentSiteAsDefault(currentSite) {
```
Comment (hmoreras) [inline src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts:194]: add type and update doc

**Review by hmoreras — CHANGES_REQUESTED** [review]

File: src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts (hunk 8 above)
Comment (fmontes) [reply src/app/view/components/_common/dot-site-selector/dot-site-selector.component.ts:172]: No, just used the `paginationService.totalRecords` directly.

**Pull request merged**